orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -165,27 +165,41 @@ public TaskResult execute(@Nonnull Stage stage) {
 
     List<Map> imagesToRestore = new ArrayList<>();
     for (String region : stageData.regions) {
-      List<ServerGroup> serverGroupsInRegion =
+      List<ServerGroup> allServerGroupsInRegion =
           serverGroups.stream()
               .filter(s -> region.equalsIgnoreCase(s.region))
               .collect(Collectors.toList());
 
-      if (serverGroupsInRegion.isEmpty()) {
-        // no server groups in region, nothing to rollback!
+      if (allServerGroupsInRegion.size() < 2) {
+        // need at least one server group to rollback from, and one to rollback to!
+        logger.warn(
+            "Not enough server groups in cluster {} and region {} to perform a rollback. Skipping this region.",
+            moniker.get().getCluster(),
+            region);
         continue;
       }
 
-      ServerGroup newestEnabledServerGroupInRegion =
-          serverGroupsInRegion.stream()
+      List<ServerGroup> enabledServerGroupsInRegion =
+          allServerGroupsInRegion.stream()
               .filter(s -> s.disabled == null || !s.disabled)
-              .findFirst()
-              .orElse(null);
+              .collect(Collectors.toList());
 
-      if (newestEnabledServerGroupInRegion == null) {
-        // no enabled server groups in this region, nothing to rollback!
+      if (enabledServerGroupsInRegion.isEmpty()) {
+        // no enabled server groups in this region, nothing to rollback from!
+        logger.warn(
+            "No enabled server groups in cluster {} and region {} to rollback from. Skipping this region.",
+            moniker.get().getCluster(),
+            region);
         continue;
       }
 
+      ServerGroup newestEnabledServerGroupInRegion = enabledServerGroupsInRegion.get(0);
+      boolean disregardDisabledCandidates =
+          stageData.additionalRollbackContext != null
+              && stageData.additionalRollbackContext.containsKey("disregardDisabledCandidates")
+              && ((Boolean) stageData.additionalRollbackContext.get("disregardDisabledCandidates"))
```

Comment:
i think you can probably do:
`&& ((Boolean) stageData.additionalRollbackContext.getOrDefault("disregardDisabledCandidates", false))`

Reply:
Yeah, this context switching between Java and Kotlin is getting to me! 😄